sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -82,8 +82,11 @@ case class CreateFunctionCommand(
     } else {
       // Handles `CREATE OR REPLACE FUNCTION AS ... USING ...`
       if (replace && catalog.functionExists(func.identifier)) {
-        // alter the function in the metastore
-        catalog.alterFunction(func)
```

Comment:
Instead of changing CreateFunctionCommand, should we change the behavior at HiveExternalCatalog.alterFunction? If this is for Hive only?

Reply:
HiveExternalCatalog is for Hive only, currently alter function of Hive does not update the resource

Reply:
@viirya do I need to handle any other cases for this PR ?

Reply:
Yeah, I think @viirya's point is valid. Isn't the root cause that `HiveExternalCatalog.alterFunction` does not handle? 

> If this is for Hive only?

I think he meant the behaviours at `InMemoryCatalog.alterFunction`.

Reply:
I got it, yes this problem is only for HiveExternalCatalog, I will update the code and push